tests/cli.rs

```diff
@@ -211,67 +215,127 @@ async fn can_run_network() {
 async fn client_without_servers_fails() {
     let (_, rpc) = start_network(2).await;
 
-    let recipient = gen_client_cmd(vec!["config", "new", &rpc.to_string()])
-        .pipe(gen_client_cmd(vec!["config", "get-public-key"]))
+    let recipient = cmd!(CLIENT_BIN, "config", "new", &rpc.to_string())
+        .pipe(cmd!(CLIENT_BIN, "config", "get-public-key"))
         .read()
         .expect("recipient public key");
 
-    gen_client_cmd(vec!["config", "new", &rpc.to_string()])
-        .pipe(gen_client_cmd(vec!["send-asset", &recipient, "10"]))
+    cmd!(CLIENT_BIN, "config", "new", &rpc.to_string())
+        .pipe(cmd!(CLIENT_BIN, "send-asset", "1", &recipient, "10"))
         .run()
         .expect_err("send asset");
 }
 
+fn get_balance(config: String) -> usize {
+    cmd!(CLIENT_BIN, "get-balance")
+        .stdin_bytes(config)
+        .read()
+        .expect("get asset")
+        .parse::<usize>()
+        .expect("parse asset amount as usize")
+}
+
 #[tokio::test]
 async fn new_client_has_some_asset() {
     let (_servers, rpc) = start_network(3).await;
 
-    let amount = gen_client_cmd(vec!["config", "new", &rpc.to_string()])
-        .pipe(gen_client_cmd(vec!["get-balance"]))
+    let config = cmd!(CLIENT_BIN, "config", "new", &rpc.to_string())
         .read()
-        .expect("get asset")
-        .parse::<u64>()
-        .expect("parse asset amount as u64");
+        .expect("create sender");
 
-    assert!(amount > 0);
+    assert!(get_balance(config) > 0);
 }
 
-#[tokio::test]
-async fn can_send_asset() {
-    fn get_balance(config: String) -> u64 {
-        gen_client_cmd(vec!["get-balance"])
-            .stdin_bytes(config)
-            .read()
-            .expect("get asset")
-            .parse::<u64>()
-            .expect("parse asset amount as u64")
+fn transfer(
+    sender_config: String,
+    sender_sequence: sieve::Sequence,
+    receiver_config: String,
+    amount: usize,
+) {
+    let second_client = cmd!(CLIENT_BIN, "config", "get-public-key")
+        .stdin_bytes(receiver_config)
+        .read()
+        .expect("get public key");
+
+    cmd!(
```

Comment:
I guess this is related to what we discussed in the meeting today: if I understand correctly, this is an integration test written in rust. My preference for the integration tests with CLIs is to use (ba)sh, because:

- it shows you how good the CLI actually is: do the parameters make sense, is it +- easy to use it?
- you don't rely on things that are only available in the rust-library, but you actually need to export everything that is needed for the user

Reply:
Right, it tried doing that but there was some issues
* integrating it with `cargo test` was hard as there is no way to generate individual test cases per file
* using `make` is weird as only the `test` target would be relevant, the rest being only duplicate of `cargo ...`

Hopefully, when doing the next project, we can directly `use test::*` https://doc.rust-lang.org/test

Reply:
Perhaps we can discuss this further tomorrow, together with @cgrigis . But after our Monday discussion I think a `run_integration_tests.sh` would be something viable. There is most probably a better (ba)sh testing framework than https://github.com/dedis/cothority/blob/main/libtest.sh !

Reply:
Face2Face: let's do it for the next tests